products/views.py

```diff
@@ -32,4 +33,29 @@ def get(self, request):
             'images':[image.url for image in product.image_set.all()]
         }for product in products]
 
-        return JsonResponse({'results': results}, status=200)  
\ No newline at end of file
+        return JsonResponse({'results': results}, status=200)  
+
+class ProductDetailView(View):
+    def get(self, request, product_id):
+        
+        if not (Product.objects.filter(id=product_id).exists()):
+            return JsonResponse({'MESSAGE': 'NOT_FOUND'}, status=404)
+
+        product = Product.objects.annotate(likes=Count("userproductlike")).get(id=product_id)
+        
+        results = {
+            'id':product.id,
+            'name':product.name,
+            'price':int(product.price),
+            'like': product.likes,
+            'images':[image.url for image in product.image_set.all()],         
+            }
+        reviews= [{
+            "user_name":comment.user.name,
+            "rate": int(comment.rate),
+            "text": comment.text,
+            "created_at":comment.created_at.date(),
+            "images":[image.url for image in comment.commentimage_set.all()]
+            }for comment in product.comment_set.all()]    
```

Comment:
Align 맞춰주세요~ 그리고 반환하는 데이터의 계층 구조를 명확하게 만들어주세요!


```suggestion
        results = {
            'id'   : product.id,
            'name' : product.name,
            'review' : [{
                "user_name" : comment.user.name,
                "rate"   : int(comment.rate),
                "text": comment.text
                ]
            ...
            }
```

Reply:
반영했습니다.